### PR TITLE
Pet of a given type

### PR DESCRIPTION
**Pull request by Jessica-Mach**
User Story: Pets of a Given Type

- set up a schema for adoptable pets
- updated the Seeder to include adoptable pets
- added adoptable pets to the data.json
- set up an api endpoint `/api/v1/petTypes/pets/:type`
- when a user clicks on a type of pet, the user gets all pets of that type
- the list of pets shows name, picture, age, and vaccination status
- styling will be treated as a separate feature

### CODE REVIEW

File: client/components/AdoptablePetsOfAType.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  }, [])
+
+
+  return (
+    <div>
+      <h1>Adoptable {petType.type}</h1>
+      {pets}
+    </div>
+  )
+
```
Comment (bkincart) [inline client/components/AdoptablePetsOfAType.js]: please remove any extra whitespace lines here!

File: client/components/AdoptablePetsOfAType.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      console.log(responseBody);
+      setPetType(responseBody.petType);
+    } catch (error) {
+      console.error(`error: ${error.message}`);
+    }
+  }
+
+  const pets = petType.adoptablePets.map( pet => {
+    return <PetTile
+      key = {pet.id}
```
Comment (bkincart) [inline client/components/AdoptablePetsOfAType.js]: indent lines 30-32

File: client/components/PetTile.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+const PetTile = (props) => {
+  const { id, name, imageUrl, age, vaccinationStatus } = props.pet;
+  return (
+    <div >
+      <Link to={`/pets/${props.type}/${id}`}>{name}</Link>
+      <Link to={`/pets/${props.type}/${id}`} >
+        <img src={imageUrl} alt={props.type} />
+      </Link>
+      <p>Age: {age}</p>
+      <p>Vaccination Status: {vaccinationStatus? 'YES' : 'NO'}</p>
```
Comment (bkincart) [inline client/components/PetTile.js]: Nice ternary :) for clarity, let's add a space between `vaccinationStatus` and the question mark!

File: server/Models/PetType.js [hunk 4]
```diff
@@ -37,6 +37,23 @@ class PetType {
             throw(err);
         }
     }
+
+    async adoptablePets(){
```
Comment (bkincart) [inline server/Models/PetType.js:41]: BEAUTIFUL association here :)

File: server/db/Seeder.js [hunk 5]
```diff
@@ -10,18 +10,28 @@ class Seeder {
   static async seed() {
     let loadedData = fs.readFileSync('data.json');
     let contents = JSON.parse(loadedData);
-    
+
     //console.log(contents.data.petTypes[0]);
```
Comment (bkincart) [inline server/db/Seeder.js]: Let's remove this extra console.log at this point :)

**Review by bkincart — CHANGES_REQUESTED** [review]
Really beautiful work here! A few small changes then okay to merge.

**Comment by Jessica-Mach** [comment]
Made the Following Changes Based on Feedback:

- removed extra spaces from AdoptablePetsOfAType.js
- indented lines 30-32 in AdoptablePetsOfAType.js
- added a space to the ternary in PetTile.js
- removed console.logs from the Seeder

**Pull request merged**